trillian/migrillian/core/multi.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"sync"
+
+	"github.com/golang/glog"
+)
+
+// RunMigration migrates data from a number of CT logs to Trillian. Each log's
+// migration is coordinated by the corresponding Controller. This function
+// Terminates when all Controllers are done (possibly with an erorr, or as a
+// result of canceling the passed in context).
+//
+// TODO(pavelkalinnikov):
+// - Expose status of each goroutine to metrics.
+// - Deal with Controller failures, e.g. cancel other Controllers and exit.
+// - Introduce a MultiController type.
```

Comment:
More TODOs for `MultiController`:
- starting `Controllers` with random delays to prevent one instance (the one that has started first) getting mastership over all logs at once
- adding voluntary mastership resignations

Reply:
Added to the list.